src/IssueViz/Editor/LineHashCalculator.cs

```diff
@@ -0,0 +1,72 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.ComponentModel.Composition;
+using SonarLint.VisualStudio.Core.Helpers;
+
+namespace SonarLint.VisualStudio.IssueVisualization.Editor
+{
+    public interface ILineHashCalculator
+    {
+        string Calculate(string wholeText, int line);
```

Comment:
`documentText` might be clearer than `wholeText`

Is `line` 0-based or 1-based? I think we should start being more explicit about this to avoid confusion or having to check in the code.
Maybe a doc comment, or we could make it explicit in the parameter name e.g. `oneBasedLineNumber`.